components/clarity-lsp/src/common/state.rs

```diff
@@ -297,10 +294,19 @@ pub async fn build_state(
     // expect contracts to be created, edited, removed.
     // A on-disk deployment could quickly lead to an outdated
     // view of the repo.
-    let manifest = ProjectManifest::from_location(manifest_location)?;
+    let manifest = match file_accessor {
+        None => ProjectManifest::from_location(manifest_location)?,
+        Some(file_accessor) => {
+            // We should create a new async function in `ProjectManifest` that can take a `file_accessor: &Box<dyn FileAccessor>`
+            // as an argument, and that will use:
+            // file_accessor.read_manifest_content()
```

Comment:
Is this what we ended up doing? Should we remove / update this comment? Explaining the rationale could be helpful for maintenance / knowledge sharing, as a lot of thoughts and experiments went into the final approach.